sycl/doc/dev/RuntimeTestingGuidelines.md

```diff
@@ -0,0 +1,123 @@
+# SYCL Runtime and Device Headers Testing Guidelines
+
+## SYCL Host Runtime Testing
+
+Most changes to the SYCL Host Runtime are device-agnostic, thus can and must be
+tested in isolation from the rest of the SYCL toolchain. Below is the
+information on some techniques, that allow one to isolate these parts of the
+runtime and simulate different behavior of low-level runtimes.
+
+### Plugin Interface Mock
+
+PI Mock layer provides capabilities to override plugins' behavior. There are two
+ways to enable PI Mock in your unit test:
+
+1. Include `helpers/sycl_test.hpp` header and use `SYCL_TEST` macro instead of
+   `TEST` macro from Google Test framework. All other Google Test macros remain
+   unchanged.
+2. Include `helpers/sycl_test.hpp` header and derive from SYCLUnitTest class
+   template. Then use the derived class with Google Test `TEST_F` macro. Make
+   sure to call base class `SetUp()` and `TearDown()` methods in child's
+   overrides.
+
+By default, PI Mock will add one plugin for each backend and set up some
+sensible overrides, enough to emulate a basic SYCL application. Each plugin will
+contain one platform with three devices in each platform: a CPU, a GPU, and an
+accelerator. You can use SYCL device selectors to pick a particular device.
+Other plugin routines will be replaced with dummy implementation, that
+terminates application immediately.
+
+Plugin Interface routines can be overriden for all backends with `redefine` free
+funcion or for a particular backend with `redefineOne` free function. Both
+functions accept a function pointer or a lambda. These overrides can be used
+to check, that a particular function has been called, or to emulate behavior
+of a low-level runtime.
+
+> If a pull request changes the way runtime plugin calls PI functions, it must
+> also contain a unit test with PI mock to verify these changes.
+
+> If a pull request adds new API, that may throw an exception, it must also
+> contain a unit test with exception path covered. In case exception depends
+> on low-level runtime call result, use PI Mock to simulate faulty call.
```

Comment:
Is it planned section? Looks like a duplication of what is stated in `CONTRIBUTING.md`.